### PR TITLE
[TensorExpr] Add plumbing for conv2d fusion.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54605 [TensorExpr] Add aten::matmuls to TE fuser.
* **#54439 [TensorExpr] Add plumbing for conv2d fusion.**

For now the only way to represent conv2d in TE is via an external call,
and since aten library doesn't have an out variant for conv2d, the
external call has to perform an extra copy. Because of that fusing
conv2d now regressed performance and hence is disabled. However, in near
future we should have two alternative ways to enable it:
1) represent conv2d natively in TE (without an external call)
2) add an out variant for conv2d

Differential Revision: [D27237045](https://our.internmc.facebook.com/intern/diff/D27237045)

### CODE REVIEW

**Review by bertmaher — APPROVED** [review]

File: torch/csrc/jit/tensorexpr/loopnest.cpp [hunk 1]
```diff
@@ -792,11 +792,15 @@ void LoopNest::inlineIntermediateBufs(bool allow_duplicated_work) {
       // tensors, always inline, bc we are not duplicating any work
       // and avoiding an intermediary buffer
       if (stores.size() == 1) {
-        auto store = dynamic_cast<Store*>(stores[0].s);
-        auto input_as_load = dynamic_cast<const Load*>(store->value());
-        if (input_as_load && input_bufs.count(input_as_load->buf())) {
-          bufs_to_inline.insert(buf);
-          continue;
+        if (auto store = dynamic_cast<Store*>(stores[0].s)) {
```
Comment (asuhan) [inline torch/csrc/jit/tensorexpr/loopnest.cpp:795]: How is this code related to the rest, is it for lowering with external calls?

**Review by asuhan — APPROVED** [review]

File: torch/csrc/jit/tensorexpr/loopnest.cpp (hunk 1 above)
Comment (ZolotukhinM) [reply torch/csrc/jit/tensorexpr/loopnest.cpp:795]: It's a fix for a bug in inliner related to ExternalCalls handling. I've added a test for this!
